### PR TITLE
Add dynamic notch settings to UI

### PR DESCRIPTION
**Pull request by IvoFPV**
Adds dynamic notch settings to the configurator PID tuning filters tab below gyro notch filters. Requires MSP which will go with this PR.

### CODE REVIEW

**Comment by McGiverGim** [comment]
One suggestion: if you can, when pushing PR to the Configurator, please paste an screen capture of the final result. Is easier review the code if we can see how it reflects in the application.

**Comment by IvoFPV** [comment]
image

**Comment by McGiverGim** [comment]
The pity about this is that it breaks the paradigm of the filter page. All of them have a check to enable or disable each filter. The Dyn Notch Filter is a feature so it is enabled/disabled in the features page. 

@mikeller is there an easy way to change the dyn filter in the firmware for version 4.1 to be like the others filters?

**Comment by IvoFPV** [comment]
@McGiverGim Problem is that dyn notch is the filter that shouldn't just be enabled or disabled like others. It is for 90% of the users always enabled or they have a really clean quad where they disable it as a feature. I'm also hiding the dyn notch settings when it is disabled.

**Comment by McGiverGim** [comment]
Yes, I don't say it must be disabled. Of course it must be enabled by default. Only that the checkbox must be here and not in the features tab.
If the user disables it we can show a warning message, the same that happens now it the user disables too much filters.

**Comment by IvoFPV** [comment]
We can easily display a warning message in the filters tab that it has been disabled. If it could be toggled from there it would be nice but this will work just fine for most people, others will just disable it in features and thats about it.

**Comment by mikeller** [comment]
@McGiverGim: Not sure if we can make the dynamic filter work exactly like the other filters on this tab, but we certainly can move the feature toggle for this from the configuration tab to the filters tab, see https://github.com/betaflight/betaflight-configurator/blob/ca5334259507c613c52e8b10a23ed615b6ab946f/src/js/tabs/failsafe.js#L241 for an example of how this is done.

**Comment by McGiverGim** [comment]
@mikeller changing a feature will need to reboot? This tab does not have a reboot and I prefer not modifying that if needed.

**Comment by mikeller** [comment]
@McGiverGim: Erg, good point, this is a drawback. At a pinch we could make the reboot conditional, and only do 'Save & Reboot' if the feature has been changed, and 'Save' in all other cases.

**Comment by IvoFPV** [comment]
Should I put the switch there and make the reboot only happen when it gets changed then? It would be nice as @McGiverGim says it won't break the paradigm of the filters page.

**Comment by mikeller** [comment]
> Should I put the switch there and make the reboot only happen when it gets changed then? 

Not sure, what do you think @McGiverGim?

**Comment by McGiverGim** [comment]
@mikeller I've been looking at the dynamic filter, and it seems to me that there is not need to reboot to make it working. The firmware part is not my strong part... Can you look at it? The features need the reboot for some thing?

**Comment by McGiverGim** [comment]
We will need to remove it manually from the features page too. Another solution is to remove this feature and enable it with some var, but I suppose this will be bad for backwards compatibility?

**Comment by mikeller** [comment]
@McGiverGim: Yes, enabling a feature requires a restart for the feature to be enabled, so if we want to move the 'DYNAMIC_FILTER' feature into the filter tab (which makes sense), we will have to offer the user a 'Save & Reboot' if the feature setting has been changed.

**Comment by IvoFPV** [comment]
I am thinking of leaving it like this for now. Currently I am working on revamping the whole PID tuning tab, so I could handle this issue as part of that project instead where I would also like to move anti-gravity as well.

**Comment by mikeller** [comment]
@IvoFPV: We could do that. But in this case, if the dynamic filter feature is disabled, I would at least put a message in its place telling the user that they have to enable the feature to be able to configure the dynamic filter, or I suspect we will get a lot of issues opened because of this.

**Comment by IvoFPV** [comment]
@mikeller When it is disabled related settings get hidden anyways, but we could add another note if it is disabled.

**Comment by mikeller** [comment]
@IvoFPV:

> When it is disabled related settings get hidden anyways,

That's the point - we'll have users ask 'why does this not show for me'.

File: locales/en/messages.json [hunk 1]
```diff
@@ -3120,6 +3120,39 @@
...
         "message": "The Notch Filter has a Center and a Cutoff. The filter is symmetrical. The Center Frequency is the center of the filter and the Cutoff Frequency is where Notch filter starts. For example with Notch Cutoff of 160 and Notch Center of 260 it means the range is 160-360Hz with most attenuation around center"
     },
+    "pidTuningDynamicNotchFilterGroup": {
+        "message": "Dynamic Notch Filter"
+    },
+    "pidTuningDynamicNotchFilterHelp": {
+        "message": "Dynamic Notch Filter tracks peak motors noise frequency and places one or two notch filters with their center at that frequency."
+    },
+    "pidTuningDynamicNotchFilterDisabledWarning": {
+        "message": "<span class=\"message-negative\">Warning:</span> Dynamic Notch Filter is disabled, please enable it as a feature in the Configuration tab to show related settings."
```
Comment (mikeller) [inline locales/en/messages.json]: This gives the impression that the user **has to** enable the dynamic notch filter, which is not what I think this should be. I'd rather go for a notice like:

Notice: The dynamic notch filter is disabled. In order to configure and use it, please enable the 'DYNAMIC_FILTER' feature in the 'Other Features' section of the 'Configuration' tab.

File: src/js/tabs/pid_tuning.js [hunk 2]
```diff
@@ -1785,9 +1819,15 @@ TABS.pid_tuning.updateFilterWarning = function() {
     var dtermDynamicLowpassEnabled = $('input[id="dtermLowpassDynEnabled"]').is(':checked');
     var dtermLowpass1Enabled = $('input[id="dtermLowpassEnabled"]').is(':checked');
     var warning_e = $('#pid-tuning .filterWarning');
+    var warningDynamicNotch_e = $('#pid-tuning .dynamicNotchWarning');
     if (!(gyroDynamicLowpassEnabled || gyroLowpass1Enabled) || !(dtermDynamicLowpassEnabled || dtermLowpass1Enabled)) {
         warning_e.show();
     } else {
         warning_e.hide();
     }
+    if (FEATURE_CONFIG.features.isEnabled('DYNAMIC_FILTER')) {
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js]: This should check the API version as well.

File: locales/en/messages.json (hunk 1 above)
Comment (IvoFPV) [reply locales/en/messages.json]: Yeah, changed it...

File: locales/en/messages.json [hunk 3]
```diff
@@ -3120,6 +3120,39 @@
...
         "message": "The Notch Filter has a Center and a Cutoff. The filter is symmetrical. The Center Frequency is the center of the filter and the Cutoff Frequency is where Notch filter starts. For example with Notch Cutoff of 160 and Notch Center of 260 it means the range is 160-360Hz with most attenuation around center"
     },
+    "pidTuningDynamicNotchFilterGroup": {
+        "message": "Dynamic Notch Filter"
+    },
+    "pidTuningDynamicNotchFilterHelp": {
+        "message": "Dynamic Notch Filter tracks peak motors noise frequency and places one or two notch filters with their center at that frequency."
+    },
+    "pidTuningDynamicNotchFilterDisabledWarning": {
+        "message": "<strong>Notice:</strong> The dynamic notch filter is disabled. In order to configure and use it, please enable the 'DYNAMIC_FILTER' feature in the 'Other Features' section of the 'Configuration' tab."
```
Comment (mikeller) [inline locales/en/messages.json]: In order to make the life of our translators easier, and to ensure consistency, you can replace the tab / section names with references to the existing translations, like `$t(configurationFeatures.message)`.

Comment (IvoFPV) [reply locales/en/messages.json]: Fixed

**Comment by McGiverGim** [comment]
Yes, I suppose we can merge it as is by the moment.

@IvoFPV I have seen that you plan to modify the PIDs/Filter tab. I'm not to sure what is your idea, but if I can made a suggestion, I think is a good idea to split the PID & Rates into two different places. In this way we can add more elements (like D_Min next to the D term) to the PID table that it has more sense. I don't know if this is what you have in mind.

@mikeller To me, the perfect solution will be change the dynamic filter from features to some more standard property in CLI in the firmware. Looking at the dynamic filter code I don't see any place where it needs a restart to be enabled/disabled. So the question is: is acceptable change it? I don't know if since a point of view of backwards compatibility is a good idea, because a feature with a known code is going to disappear.

**Comment by mikeller** [comment]
@McGiverGim: Yes, I think we need to look into changing from features to standardised parameters at some point - there is not much point in having special logic for features, and some of them have even '1 of x options' logic, making them completely unintuitive as features.

**Pull request merged**